### DISCUSSION TITLE
Improving smoke visualisation

### DISCUSSION THREAD

**Discussion by xingjianhou** [Part-II-Computational-Physics/.github#99] [General]
For the Schrodinger's smoke project, I was able to advect the particles to show their positions (in the first figure). However, many details of the smoke, especially the vortex structures, are not well-resolved. Any suggestions on how to efficiently improve visualisation using Python? I hope to do as well as in the second figure.
![image](https://user-images.githubusercontent.com/73950854/231853738-b4bb4d99-349f-4bee-bda0-8cfd1600bbfd.png)
![image](https://user-images.githubusercontent.com/73950854/231852557-8c5e08df-8791-4ea0-a6dc-7245bdf9ae0f.png)

**Comment by AustenLamacraft** [comment]
Looks nice already! If you want to get more into 3d visualisation. You could check out [Mayavi](https://docs.enthought.com/mayavi/mayavi/), [pyvista](https://docs.pyvista.org/version/stable/index.html),  [pythreejs](https://github.com/jupyter-widgets/pythreejs), or [ipyvolume](https://ipyvolume.readthedocs.io/en/latest/). The first two are based on VTK and the second two on WebGL. 

You might want to prepare the visualizations in a separate notebook or file and then generate images to be embedded in the report notebook, so you won't need to rely on these libraries for the report itself.
